PHPUnit tests for ElggFile

hypeJunction opened the pull request:
**chore(tests): adds tests for common file handle operations**

Adds PHPUnit tests for common file handle operations. Also tests if ElggFile
can follow and handle symlinks

**fix(file): ElggFile::delete() now removes target files if filename is a symlink**

By default, ElggFile::delete() now also deletes the target of a symbolic link,
if the filename is a link.
- [x] Randomize filenames

hypeJunction committed:
chore(tests): adds tests for common file handle operations

Adds PHPUnit tests for common file handle operations. Also tests if ElggFile
can follow and handle symlinks

hypeJunction committed:
fix(file): ElggFile::delete() now removes target files if filename is a symlink

By default, ElggFile::delete() now also deletes the target of a symbolic link,
if the filename is a link.

hypeJunction commented:
@mrclay Could you take a look at ``ElggFile::seek()``. It's either a bug, or I am misunderstanding what it is supposed to do, I have marked the test as skipped for now.

mrclay commented on engine/classes/ElggFile.php:
```diff
@@ -294,12 +294,13 @@ public function close() {
 	 * @return bool
 	 */
-	public function delete() {
+	public function delete($follow_symlinks = true) {
```
We don't get an E_STRICT error here, from not matching ElggEntity?

hypeJunction replied:
I thought about that, but looks like not.
On Apr 9, 2016 9:20 PM, "Steve Clay" <notifications@github.com> wrote:

> In engine/classes/ElggFile.php
> <https://github.com/Elgg/Elgg/pull/9647#discussion_r59119959>:
>
> >  	 * @return bool
> >  	 */
> > -	public function delete() {
> > +	public function delete($follow_symlinks = true) {
>
> We don't get an E_STRICT error here, from not matching ElggEntity?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/Elgg/Elgg/pull/9647/files/facc13fe13f8719204c85b7f7334acfa5e8f06da#r59119959>
>

mrclay replied:
Nevermind. Subclass signatures don't have to match if we're just adding param(s).

mrclay committed:
docs(core): fixes docs for ElggFilestore::seek return value

mrclay committed:
chore(tests): fixes ElggFile::seek test

mrclay commented:
Added some commits https://github.com/hypeJunction/Elgg/pull/3

hypeJunction commented:
@mrclay merged your commits

mrclay commented:
> Randomize filenames

Maybe add an issue for this and we can move this PR forward?

mrclay commented:
LGTM

hypeJunction commented:
A failing test would leave artefacts behind. I will change the owner and wipe the dir on setup and teardown. Will do that tomorrow

hypeJunction committed:
chore(tests): avoid name collissions in file tests

Tests now use a different owner_guid for written files, and clean up
owner directory before and after the test

hypeJunction commented:
@mrclay I updated the tests to use a different owner wiping the owner directory before and after the tests.
